File: scala-pr/54/scala/code-quality-error-handling/SecureDatabaseConnector.scala
import java.sql.{Connection, DriverManager, SQLException}
import scala.util.{Try, Success, Failure}

class SecureDatabaseConnector {
  private val host = "localhost"
  private val database = "testdb"
  private val username = "root"
  private val password = "password"

  def connect(): Try[Connection] = {
    // {fact rule=code-quality-error-handling@v1.0 defects=0}
    Try {
      DriverManager.getConnection(s"jdbc:mysql://$host/$database", username, password)
    } match {
      case Success(connection) => Success(connection)
      case Failure(ex: SQLException) =>
        println(s"Database connection failed: ${ex.getMessage}")
        Failure(new RuntimeException(s"Unable to connect to database: ${ex.getMessage}", ex))
      case Failure(ex) =>
        println(s"Unexpected error: ${ex.getMessage}")
        Failure(ex)
    }
    // {/fact}
  }
}

File: kotlin-pr/50/kotlin/code-quality-error-handling/SecureDatabaseConnector.kt
import java.sql.Connection
import java.sql.DriverManager
import java.sql.SQLException

class SecureDatabaseConnector {
    private val host = "localhost"
    private val database = "testdb"
    private val username = "root"
    private val password = "password"

    fun connect(): Result<Connection> {
        // {fact rule=code-quality-error-handling@v1.0 defects=0}
        return try {
            val connection = DriverManager.getConnection("jdbc:mysql://$host/$database", username, password)
            Result.success(connection)
        } catch (ex: SQLException) {
            println("Database connection failed: ${ex.message}")
            Result.failure(RuntimeException("Unable to connect to database: ${ex.message}", ex))
        } catch (ex: Exception) {
            println("Unexpected error: ${ex.message}")
            Result.failure(ex)
        }
        // {/fact}
    }
}